src/core/services/compiler/compiler.js

```diff
@@ -133,9 +133,17 @@ function MdCompilerProvider($compileProvider) {
    * Material Dialog/Panel/Toast/BottomSheet controllers using the `$controller` helper
    * as it always follows the `$compileProvider.preAssignBindingsEnabled()` value.
    */
-  // TODO change it to `true` in Material 1.2.
-  var respectPreAssignBindingsEnabled = false;
+  var respectPreAssignBindingsEnabled;
+  if (angular.version.major === 1 && angular.version.minor === 7) {
+    respectPreAssignBindingsEnabled = true;
+  } else {
+    respectPreAssignBindingsEnabled = false;
+  }
   this.respectPreAssignBindingsEnabled = function(respected) {
+    if (!respected && angular.version.major === 1 && angular.version.minor === 7) {
```

Comment:
`=== 7` -> `>= 7`

Reply:
The `!respected` check means you'll catch a getter as well so this API won't work at all in 1.7. The error message suggests you meant to add this error for the setter, not a getter. The getter is probably fine.

IOW, this should be moved to inside the `if (angular.isDefined(respected))` block.